Lib9c/Action/UpdateSell.cs

```diff
@@ -118,94 +88,120 @@ public override IAccountStateDelta Execute(IActionContext context)
+                    context.BlockIndex);
+
+                // for sell cancel
+                sw.Restart();
+                if (!states.TryGetState(shopAddress, out BxDictionary shopStateDict))
+                {
+                    errors.Add((updateSellInfo.orderId,
+                        ShopErrorType.ERROR_CODE_FAILED_LOADING_STATE,
+                        shopAddress.ToHex()));
+                    continue;
+                }
+
+                sw.Stop();
+                Log.Verbose("{AddressesHex} UpdateSell Sell Cancel Get ShopState: {Elapsed}", addressesHex, sw.Elapsed);
+                sw.Restart();
+                if (!states.TryGetState(Order.DeriveAddress(updateSellInfo.orderId), out Dictionary orderDict))
+                {
+                    errors.Add((updateSellInfo.orderId,
+                        ShopErrorType.ERROR_CODE_FAILED_LOADING_STATE,
+                        Order.DeriveAddress(updateSellInfo.orderId).ToHex()));
+                    continue;
+                }
+
+                var orderOnSale = OrderFactory.Deserialize(orderDict);
+                orderOnSale.ValidateCancelOrder(avatarState, updateSellInfo.tradableId);
+                var itemOnSale = orderOnSale.Cancel(avatarState, context.BlockIndex);
+                if (context.BlockIndex < orderOnSale.ExpiredBlockIndex)
+                {
+                    var shardedShopState = new ShardedShopStateV2(shopStateDict);
+                    shardedShopState.Remove(orderOnSale, context.BlockIndex);
+                    states = states.SetState(shopAddress, shardedShopState.Serialize());
+                }
+
+                digestList.Remove(orderOnSale.OrderId);
+                states = states.SetState(itemAddress, itemOnSale.Serialize())
+                    .SetState(digestListAddress, digestList.Serialize());
+                sw.Stop();
+
+                var expirationMail = avatarState.mailBox.OfType<OrderExpirationMail>()
+                    .FirstOrDefault(m => m.OrderId.Equals(updateSellInfo.orderId));
+                if (!(expirationMail is null))
+                {
+                    avatarState.mailBox.Remove(expirationMail);
+                }
+
+                // for updateSell
+                var updateSellShopState = states.TryGetState(updateSellShopAddress, out Dictionary serializedState)
+                    ? new ShardedShopStateV2(serializedState)
+                    : new ShardedShopStateV2(updateSellShopAddress);
+
+                Log.Verbose("{AddressesHex} UpdateSell Get ShardedShopState: {Elapsed}", addressesHex, sw.Elapsed);
+                sw.Restart();
+                var newOrder = OrderFactory.Create(context.Signer, sellerAvatarAddress,
+                    updateSellInfo.updateSellOrderId, updateSellInfo.price,
+                    updateSellInfo.tradableId, context.BlockIndex, updateSellInfo.itemSubType, updateSellInfo.count);
+                newOrder.Validate(avatarState, updateSellInfo.count);
```

Comment:
Validate throw Exception. we must handle this.

Reply:
I fixed it. But I'm not sure this change was valid. Could you check it again?

Reply:
Sorry, All parts related to the review have been deleted. Could you check it again?